test/test_integration.py

```diff
@@ -115,6 +115,81 @@ def test_Spin2_backend() -> None:
     )
 
 
+def test_integration_global_phase() -> None:
+    qc = Circuit.from_string(
+        """
+        version 3.0
+
+        // This is a single line comment which ends on the newline.
+        // The cQASM string must begin with the version instruction (apart from any preceding comments).
+
+        /* This is a multi-
+        line comment block */
+
+        qubit[3] q
+
+        H q[0:2]
+        Ry(1.5789) q[0]
+        H q[0]
+        CNOT q[1], q[0]
+        Ry(3.09) q[0]
+        Ry(0.318) q[1]
+        Ry(0.18) q[2]
+        CNOT q[2], q[0]
+        """,
+    )
+
+    # Decompose 2-qubit gates to a decomposition where the 2-qubit interactions are captured by CNOT gates
+    qc.decompose(decomposer=CNOTDecomposer())
+
+    # Replace CNOT gates with CZ gates
+    qc.replace(
+        CNOT,
+        lambda control, target: [
+            H(target),
+            CZ(control, target),
+            H(target),
+        ],
+    )
+
+    # Merge single-qubit gates and decompose with McKay decomposition.
+    qc.merge_single_qubit_gates()
+    qc.decompose(decomposer=McKayDecomposer())
+
+    assert (
+        str(qc)
+        == """version 3.0
+
+qubit[3] q
+
+Rz(1.5707963) q[1]
+X90 q[1]
+Rz(1.5707963) q[1]
+Rz(3.1415927) q[0]
+X90 q[0]
+Rz(0.0081036221) q[0]
+X90 q[0]
+Rz(3.1415927) q[0]
+CZ q[1], q[0]
+X90 q[2]
+Rz(1.3907963) q[2]
+X90 q[2]
+X90 q[0]
+Rz(0.051592695) q[0]
+X90 q[0]
+Rz(3.1415927) q[0]
+CZ q[2], q[0]
+Rz(1.5707963) q[0]
+X90 q[0]
+Rz(1.5707963) q[0]
+Rz(3.1415927) q[1]
+X90 q[1]
+Rz(2.8235927) q[1]
+X90 q[1]
```

Comment:
Just out of curiosity, what instructions, if any, do the global phase correction here?